test/ngamsTestLib.py

```diff
@@ -1776,4 +1818,156 @@ def waitTillWokenUp(self, dbConObj, node, timeOut, nodes):
         self.fail("Sub-node not woken up within %ds" % timeOut)
 
 
+if ssl is not None:
+    class CAHandler(object):
+        def __init__(self, add_to_certifi=True):
+            self._create_ca()
+            if add_to_certifi:
+                self.add_ca_to_certifi()
```

Comment:
I think this is causing problems when running the tests multiple times. Each time this get executed, a new certificate is appended into the vault for the same CA; then I get `SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)')` probably due to the different certificates for the same CA? If I go to `$VIRTUAL_ENV/lib/python3.6/site-packages/certifi/cacert.pem` and remove the last certificates generated by this code I can run the test again.

I think a better approach would be not to save the CA perm into the central vault, and instead capture it and give it to the ngamsPClient for usage with `requests`. It's a bit more of code, but you'd avoid touching the user's vault, and add the flexibility to the ngamsPClient to use arbitrary CA certs.